app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/AdaptResourceModelStockStatusToMultiStocks.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalog\Plugin\CatalogInventory;
+
+use Magento\Catalog\Model\ResourceModel\Product\Collection;
+use Magento\CatalogInventory\Model\ResourceModel\Stock\Status;
+use Magento\Framework\DB\Select;
+use Magento\InventoryCatalog\Model\GetStockIdForCurrentWebsite;
+use Magento\InventoryCatalog\Model\ResourceModel\AddStockDataToCollection;
+use Magento\InventoryCatalog\Model\ResourceModel\AddStockStatusToSelect;
+use Magento\Store\Model\Website;
+
+/**
+ * Adapt Resource Model Stock Status for Multi-Source Inventory.
+ */
+class AdaptResourceModelStockStatusToMultiStocks
```

Comment:
Need to split this logic in different plugins

```
<type name="Magento\CatalogInventory\Model\ResourceModel\QtyCounterInterface">
        <plugin name="plugin1" type="plugin1"/>
        <plugin name="plugin2" type="plugin2"/>
    </type>
```

Reply:
Done.